src/sidebar/util/groups.js

```diff
@@ -20,7 +25,42 @@ function combineGroups(userGroups, featuredGroups) {
   featuredGroups.forEach(group => (group.isMember = false));
   userGroups.forEach(group => (group.isMember = true));
 
-  return userGroups.concat(featuredGroups);
+  const groups = userGroups.concat(featuredGroups);
+
+  // Add isScopedToUri property indicating whether a group can be
+  // annotated in at this particular url.
+  groups.forEach(group => (group.isScopedToUri = isScopedToUri(group, uri)));
+
+  return groups;
+}
+
+function isScopedToUri(group, uri) {
+  // If the group has scope info, the scoping is enforced,
+  // and the uri patterns don't inlude this page's uri
+  // the group is not selectable, otherwise it is.
+  if (
+    group.scopes &&
+    group.scopes.enforced &&
+    group.scopes.uri_patterns &&
+    !uriMatchesScopes(uri, group.scopes.uri_patterns)
```

Comment:
This conditional has four tests in it, which makes it a little tricky to unpack quickly! A couple of thoughts and options here...

First, I want to make sure I understand what the test for truthiness of `group.scopes.uri_patterns` is meant to do. Are you intending to test here for presence of the attribute (i.e. make sure it's not `undefined`)? Or for actual truthiness? It looks like `uriMatchesScopes` is expecting an Array, not another truthy kind of value. So that's why I want to get to the bottom of what you're testing for here...

I think you can get the last test out of the conditional and, if you'd like, write this as:

```
if (
  group.scopes &&
  group.scopes.enforced &&
  group.scopes.uri_patterns) { // though a little unsure on this last test

    return uriMatchesScopes(uri, group.scopes.uri_patterns); // as this already returns a Boolean
}
```

Reply:
The truthiness I’m testing for here is whether the attribute exists. I’m not sure that’s entirely necessary? I welcome your opinion! Also I like the idea of returning the result of the uriMatchesScopes function. 🙂

Reply:
I guess a follow-up question would be: what would happen if the property was defined but was not an Array? What _should_ happen?

Reply:
And now, that question is leading me to think that this kind of validation doesn't belong in this part of the code. Because what would happen? 1) it might error if it's not iterable or 2) it could be the wrong iterable and possibly lead to a faulty match if a string was passed and it encountered a single * character. And I don't think it's unreasonable to assume that if scopes exists so do the keys inside it as that's the current implementation of the api in h.

Reply:
Anyways...long story short. I've removed it for now cause even if we want to check this I don't think it belongs in this part of the code.